ocaml/xapi/xenctrlx.ml

```diff
@@ -0,0 +1,27 @@
+
+module D = Debug.Make(struct let name = "xenctrlx" end)
+
+let handle = ref None
+
+let with_intf f =
+  match !handle with
+    | Some xc -> f xc
+    | None ->
+      let xc =
+        try
+          Xenctrl.interface_open ()
+        with
+        | e ->
+            let msg = Printexc.to_string e in
+            failwith @@ "failed to open xenctrl interface: "^msg
+      in
+        handle := Some (xc);
```

Comment:
Why not making the open lazy, and Lazy.force it in with_intf? That should also be thread safe right?

Reply:
But then it would not be nice for close

Reply:
Oh, no, it would be ok, you could use Lazy.is_val

Reply:
I believe `Xenctrl.with_intf` should implement this to avoid that every client has to do it - I will propose it. The problem there is that we have low-level `open/close` but also `with_intf` which maintains a global handle. A client should not use both methods (although it would not be dangerous).